tracing-modality/src/common/attr_handlers.rs

```diff
@@ -0,0 +1,228 @@
+//! Attribute Handlers
+//!
+//! Attribute handlers allow for project-specific modifications of `tracing` data
+//! provided when events occur.
+//!
+//! This is typically needed when data needs to be modified in some way before reporting
+//! to modality, or if a translation from "project specific" keys or values to "modality
+//! specific" keys or values are necessary.
+//!
+//! Handlers are matched on a SPECIFIC key text. If a matching key text is found, the
+//! provided handler will be called, will replace the key and value of the data reported
+//! by `tracing`.
+//!
+//! ## Default Handlers
+//!
+//! By default, the following handlers are defined:
+//!
+//! | tracing key                      | modality key                           | value conversion         |
+//! | :---                             | :---                                   | :---                     |
+//! | "timestamp"                      | "event.timestamp"                      | see [timestamp]          |
+//! | "interaction.remote_timestamp"   | "event.interaction.remote_timestamp"   | see [remote_timestamp]   |
+//! | "interaction.remote_timeline_id" | "event.interaction.remote_timeline_id" | see [remote_timeline_id] |
+//! | "name"                           | "event.name"                           | see [name]               |
+//! | "message"                        | "event.message"                        | see [name]               |
+//! | "severity"                       | "event.severity"                       | see [severity]           |
+//! | "source.module"                  | "event.source.module"                  | see [source_module]      |
+//! | "source.file"                    | "event.source.file"                    | see [source_file]        |
+//!
+//! These default handlers can be overridden using the [Options::set_attr_handlers()](crate::Options::set_attr_handlers)
+//! or [Options::with_attr_handlers()](crate::Options::with_attr_handlers) functions.
+//!
+//! ## Fallback handler
+//!
+//! Additionally, there is a "fallback handler" that will take any tracing key NOT matched by the attribute
+//! handlers by ensuring that the given key starts with "event.*", and the contained data is converted into
+//! a format usable by modality. If this conversion is suitable for data logged by your application, then no
+//! specific attribute handler needs to be provided.
```

Comment:
Can we rename this to "Default handler" just so we're not risking giving people the impression that it's a bad thing for this to not be overridden?